### PR TITLE
Adjust a bunch of .future files impacted by changes to FCFs

### PR DESCRIPTION
**Pull request by dlongnecke-cray**
Adjust a bunch of .future files impacted by changes to FCFs

Remove the .compopts for a parsing related future because it was
no longer needed. Promote two futures.

Reviewed by @DanilaFe. Thanks!

Signed-off-by: David Longnecker <dlongnecke-cray@users.noreply.github.com>

### CODE REVIEW

File: test/extern/fnPtrs/improveErrorForFcfMismatch.bad [hunk 1]
```diff
@@ -1,4 +1,4 @@
-improveErrorForFcfMismatch.chpl:13: error: unresolved call 'bar(shared chpl__fcf_type_void_void)'
```
Comment (DanilaFe) [inline test/extern/fnPtrs/improveErrorForFcfMismatch.bad:1]: Do we want this? (`chpl__` to `proc()`? Seems like it's losing some descriptiveness).

File: test/extern/fnPtrs/sugarFcfTypes2.future [hunk 2]
```diff
@@ -1,4 +1,4 @@
-error message: Need to sugar first-class function (fcf) types
-
-We shouldn't be printing out internal fcf type names when we
-generate errors involving first-class functions.
+sugarFcfTypes2.chpl:13: error: unresolved call 'barIntInt(proc(x: int): int)'
```
Comment (DanilaFe) [inline test/extern/fnPtrs/sugarFcfTypes2.future:1]: My understanding was that futures should describe the issue in human-readable format (that's what's the "before" is doing in this file). However, it seems like you've put in the "desired error message format", without any kind of description. Doesn't this stuff go into `sugarFcfTypes2.good`?

File: test/extern/fnPtrs/improveErrorForFcfMismatch.bad (hunk 1 above)
Comment (dlongnecke-cray) [reply test/extern/fnPtrs/improveErrorForFcfMismatch.bad:1]: The previous type was the underlying generated type, now it is user facing syntax for a function type.

File: test/extern/fnPtrs/sugarFcfTypes2.future (hunk 2 above)
Comment (dlongnecke-cray) [reply test/extern/fnPtrs/sugarFcfTypes2.future:1]: Hmmm, nope, somehow I think I cp'd to the wrong file name, ty!

Comment (dlongnecke-cray) [reply test/extern/fnPtrs/sugarFcfTypes2.future]: I promoted these futures.

**Pull request merged**